lstchain/scripts/lstchain_mc_dl2_to_irf.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
+
+"""
+Script to generate IRFs from MC DL2 files, using pyirf functions.
+This will save the IRFs, and either use fixed cuts or optimise them.
+and later it can be used by lstchain_dl2_to_dl3 to generate DL3 files.
+
+TODO: Add other IRFs
+Use optimised cuts and binning
+
+- Input: Path where the merged DL2 data HDF5 files are present
+- Output: IRFs compiled in fits.gz format
+
+Usage:
+$> python lstchain_mc_dl2_to_irf.py
+--input-file-gamma ./gamma/dl2_gamma_*.h5
+--input-file-gamma-diff ./gamma-diff/dl2_gamma-diff*.h5 #optional for now
+--output-dir ./IRFs/
+--pnt-like True
+--config ../../data/data_selection_cuts.json
+"""
+
+import os
+import json
+import numpy as np
+import argparse
+from pathlib import Path
+from distutils.util import strtobool
+import logging
+
+from lstchain.io import read_mc_dl2_to_pyirf, read_configuration_file, get_standard_config
+from lstchain.reco.utils import filter_events
+
+from astropy.io import fits
+import astropy.units as u
+from astropy import table
+
+# pyIRF packages
+from pyirf.io.gadf import (create_aeff2d_hdu, create_energy_dispersion_hdu)
+from pyirf.irf import (effective_area_per_energy, energy_dispersion, effective_area_per_energy_and_fov)
+from pyirf.utils import calculate_source_fov_offset, calculate_theta
+
+from pyirf.binning import (
+    create_bins_per_decade,
+    add_overflow_bins,
+    create_histogram_table,
+)
+
+log = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser(description="MC DL2 to IRF")
+
+# Required arguments
+parser.add_argument('--input-file-gamma', '-fg', type=Path, dest='gamma_file',
+                    help='Path to the DL file of point like gamma events for building IRF',
+                    default=None, required=False
+                    )
+
+parser.add_argument('--output-irf-dir', '-o', type=Path, dest='output_irf_dir',
+                    help='Path to output IRF files',
+                    default=None, required=True
+                    )
+
+#Optional arguments
+parser.add_argument('--input-file-gamma-diff', '-fg-diff', type=Path, dest='gamma_diff_file',
+                    help='Path to the dl2 file of gamma diffuse events for building IRF',
+                    default=None, required=False
+                    )
+
+parser.add_argument('--pnt-like', '-pnt', action='store',
+                    type=lambda x: bool(strtobool(x)), dest='irf_type',
+                    help='True for point-like IRF, False for Full Enclosure',
+                    default=None, required=False
```

Comment:
These two arguments are redundant.

Also, why are you storing this in `irf_type`? an `irf_type` variable that is True for point-like and False for FE is very confusing. Better just use `--point-like`.

Reply:
OK, thanks!